retrieverdash/dashboard_script/dashboard_script.py

```diff
@@ -38,19 +57,36 @@ def check_dataset(dataset):
             dataset_detail['dataset_details'] = {}
 
         if dataset_type(dataset) == 'spatial':
-            workdir = None
-            try:
-                workdir = mkdtemp(dir=file_location)
-                download(dataset.name, path=workdir)
-                md5 = getmd5(workdir, data_type='dir')
-            finally:
-                if workdir:
-                    rmtree(workdir)
+            install_postgres(dataset)
+            md5 = getmd5(path.join(file_location, 'current',
+                                   dataset.name), data_type='dir')
+
+            if dataset.name not in dataset_detail['dataset_details'] or md5 != dataset_detail['dataset_details'][dataset.name]['md5']:
+                diff = diff_generator_spatial(dataset)
+            else:
+                for keys in dataset.tables:
+                    file_name = '{}.{}'.format(
+                        dataset.name.replace('-', '_'), keys)
+                    html_file_name = '{}.html'.format(file_name)
+                    if os.path.exists(os.path.join(file_location, 'diffs', html_file_name)):
+                        remove(os.path.join(file_location,
+                                            'diffs', html_file_name))
+            data_shift(dataset, is_spatial=True)
+
         else:
             md5 = get_dataset_md5(dataset)
-            if dataset.name not in dataset_detail \
-                    or md5 != dataset_detail[dataset.name]['md5']:
+
+            if dataset.name not in dataset_detail['dataset_details'] or md5 != dataset_detail['dataset_details'][dataset.name]['md5']:
```

Comment:
This code looks repeated at L64

Reply:
I guess only the first line is repeated at lines 64 and 79, the rest of the code is actually different for tabular and spatial datasets due to different naming convention in the files. Only the check is the same.